### ISSUE TITLE
Support for ordering and pagination

### ISSUE THREAD

**Issue by kerajel** [fschick/FilterExpressionCreator#6]
Hello,

I have been working with this library and found it to be incredibly useful. However, I would like to propose an enhancement that I believe could significantly improve its usability, especially when it comes to server-side grid applications.

Currently, the library lacks support for dynamic ordering and pagination. These features are frequently utilized in modern applications and are essential for efficiently managing large data sets. Without them, it can be challenging to implement effective grid controls, leading to performance issues and a less-than-optimal user experience.

I propose the addition of the following functionalities:

Dynamic Ordering: Allow users to sort the data based on different attributes, both ascending and descending. This would include the flexibility to change the ordering criteria on the fly, adapting to user preferences.

Pagination Support: Implement robust pagination controls that enable users to navigate through large data sets easily. This includes the ability to customize the number of items per page and navigate to specific pages directly.

**Comment by fschick** [comment]
I've thought about these functions before and they shouldn't be too difficult to implement. I'm currently on vacation, but will be happy to take a look in a week or two when I get back.

**Comment by fschick** [comment]
Had a look at this. As mentioned, I'd like to have this feature too, but it's not a done in a few hours. If it's more urgent and you're able to help with this, leave a comment.

**Comment by kerajel** [comment]
Sure, there's no urgency on my end for these features. However, I'm willing to contribute if I can be of any help.

**Comment by fschick** [comment]
I thought more about the requirement. I think for the core functionality it would be the best to take the logic around the classes `EntityFilter` and `ValueFilter` as a basis and analogously create an `EntitySort` and `ValueSort`. For the OpenAPI interface, an additional parameter with the syntax `<FieldName1> <ASC|DESC>, <FieldName2> <ASC|DESC>` would be useful. Pagination feature should be easier with two additional parameters and some smaller helper classes.

As far as I can estimate, I would be able to start development at the end of October '23. Release would then probably be at the end of '23.

Basically I would also accept a pull request, but I am sure the task is tricky in detail and it will take time until everything is working and tested. If you like and be able to do this, drop a message and we can discuss some details.

**Comment by kerajel** [comment]
Thank you for considering the enhancement proposal and for sharing your thoughts on how best to implement it. I understand that the features are intricate and would require a careful and well-thought-out approach.

Given the complexities and your expertise with the library, I believe it would be best for you to handle the development of these features when you're able to start at the end of October. I trust your judgment on how to integrate these functionalities most effectively.

**Comment by fschick** [comment]
Later than expected, but I started with the [development](https://github.com/fschick/FilterExpressionCreator/tree/main-fs-sorting).

**Comment by fschick** [comment]
Ordering is done, see release [4.6.1](https://github.com/fschick/FilterExpressionCreator/releases)

Next step will be a major release include a rename of the product / repository. Afterwards I'll have a view to pagination.

**Comment by fschick** [comment]
Pagination will be tracked here #7

**Comment by fschick** [comment]
Pagination is done with version 5.0.0
